GUAC-62.5-RefactoredCodeFrontEndP1

yan-chan opened the pull request.

mosguinz commented:
Hi @yan-chan, thanks for submitting the PR. Could you please outline your changes in the PR body? Thanks!

yan-chan commented:
InventoryMenu:
- removed the duplicate API calls (in create and constructor)
- modified a bit of the UI logic and the inventory storing logic
- Remove repetition
BasicScene:
- remove duplicates/minimize repetition. 
BuildMenu:
- added a couple helper functions.
- made the adjustable values easier to manage and less hard coded
Crafter:
- added a bit of helper function
- added error handling for debugging purposes
- changed up the scope a bit so no possible confusion

mosguinz commented:
It seems that this branch should be merged into the existing GUAC-62-Refactor-the-Frontend instead. Please create a new PR to merge into there instead.
